### PR TITLE
refactor: Only publish single aws-pdk package

### LINKED ISSUE #365: [BUG] Workspace linking for constructs is not working
### Describe the bug

Consider a monorepo with one construct library

```
const root = new nx_monorepo.NxMonorepoProject({
  defaultReleaseBranch: 'main',
  devDeps: [`aws-prototyping-sdk@${AWS_PDK_VERSION}`, 'lerna@^6.6.1', '@nrwl/devkit@^15.9.2'],
  name: 'nsa',
  projenrcTs: true,
  description: 'Nsl SAAS Accelerator on AWS',
  packageName: '@nsa/aws',
  packageManager: NodePackageManager.PNPM,
  prettierOptions: {
    settings: {
      singleQuote: true,
      tabWidth: 2,
      printWidth: 120,
      trailingComma: TrailingComma.ALL,
      arrowParens: ArrowParens.AVOID,
    },
  },
  gitignore: ['.idea', '.vscode'],
  stale: true,
  autoApproveUpgrades: true,
  autoApproveOptions: {
    allowedUsernames: ['rverma-nsl'],
  },
  autoMerge: true,
  buildWorkflow: true,
  dependabot: true,
  github: true,
  minNodeVersion: '18.0.0',
  typescriptVersion: '~5.0.4',
  vscode: true,
  workspaceConfig: {
    linkLocalWorkspaceBins: true,
  },
});

new awscdk.AwsCdkConstructLibrary({
  parent: root,
  outdir: 'packages/constructs',
  bundledDeps: [
    '@nslhb/eks-blueprints@1.6.21',
    `@aws-sdk/client-iam@${AWS_SDK_VERSION}`,
    `@aws-sdk/client-secrets-manager@${AWS_SDK_VERSION}`,
    '@types/js-yaml@4.0.5',
    'js-yaml@4.1.0',
    'sync-request',
  ],
  peerDeps: ['cdk-nag'],
  devDeps: ['@types/aws-lambda', 'aws-lambda', 'jsii-pacmak@1.80.0' // need to ovveride this, default is very old inaccessible version],
  defaultReleaseBranch: 'main',
  name: '@nsa/construct',
  packageManager: NodePackageManager.PNPM,
  cdkVersion: CDK_VERSION,
  constructsVersion: CONSTRUCT_VERSION,
  minNodeVersion: root.minNodeVersion,
  jestOptions: {
    jestVersion: JEST_VERION, // need to override this as using ts 5.0.4
    jestConfig: {
      detectOpenHandles: true,
    },
  },
  lambdaOptions: {
    runtime: awscdk.LambdaRuntime.NODEJS_18_X,
    bundlingOptions: {
      externals: ['aws-sdk'],
      sourcemap: true,
    },
  },
  jsiiVersion: '5.0.6',
});
```
Task "build » package" failed when executing "pdk@pnpm-link-bundled-transitive-deps

### Expected Behavior

Successful build.
Doesn't require a manual copy of the scripts from nx-mono-repo project.

### Current Behavior

```
@nsa/construct: 👾 build » package | pdk@pnpm-link-bundled-transitive-deps packages/constructs
@nsa/construct: /bin/sh: pdk@pnpm-link-bundled-transitive-deps: command not found
@nsa/construct: 👾 Task "build » package" failed when executing "pdk@pnpm-link-bundled-transitive-deps packages/constructs" (cwd: /Users/rverma/dev/aws/nsa/packages/constructs)
@nsa/construct:  ELIFECYCLE  Command failed with exit code 1.
```

### Reproduction Steps

simply use above projen.ts file and build any construct

### Possible Solution

Had to copy the scripts from https://github.com/aws/aws-prototyping-sdk/tree/mainline/packages/nx-monorepo/scripts/pnpm to my construct subproject manually.

### Additional Information/Context

_No response_

### PDK version used

0.17.0

### What languages are you seeing this issue on?

Typescript

### Environment details (OS name and version, etc.)

mac m1

### LINKED ISSUE #527: [BUG] cdk-graph-plugin-diagram has issues in Py/Java
### Describe the bug

1.) Some methods like `pruneCustomResources` are not available given these are defined as static functions cannot be exported by JSII. These missing functions should be encapsulated in a class in order to expose them properly.

2.) The existing callback functions do not work in PY/JAVA. These will need to be augmented with cross-language compatible ones. For example if we consider [`IGraphFilterPlan`](https://github.com/aws/aws-prototyping-sdk/blob/mainline/packages/cdk-graph/src/filtering/types.ts#L112C18-L112C34) which has member `filters?: (IGraphFilter | IGraphStoreFilter)[];`. The `IGraphStoreFilter` is defined as a function which is not supported in other languages and hence this filters may potentially need to be augmented with something like: `filters?: (IGraphFilter | IGraphStoreFilter | IGraphStoreFilterVerbose)[];`

```
interface IGraphStoreFilterVerbose {
  filter(store: Graph.Store): void;
}
```  

3.) Certain accessors do not seem to be present in the exposed interfaces. For example [`parent()`](https://github.com/aws/aws-prototyping-sdk/blob/mainline/packages/cdk-graph/src/core/graph.ts#L1131) on the Node class is not exposed even though it should be. This may need to have supplementary public getters defined if typescript accessors are not supported.

### Expected Behavior

1.) These methods should be available in all languages.

2.) Callback functions should work in all available languages.

3.) Public properties/methods should be available in all languages.

### Current Behavior

Same as above.

### Reproduction Steps

N/A

### Possible Solution

As Above.

### Additional Information/Context

_No response_

### PDK version used

latest

### What languages are you seeing this issue on?

Java, Python

### Environment details (OS name and version, etc.)

macos

### PR DESCRIPTION
**Pull request by agdimech**
### Changes

This PR restructures the PDK as follows:

- Individual packages are now namespaced as `@aws-pdk` and are **NOT** published to npm/pypi/maven.
- A new `aws-pdk` package is created which creates a single distributable which **IS** published to npm/pypi/maven.
- Dependencies between packages is supported via `@aws-pdk/XXX`. When packaging into the `aws-pdk`, these imports are rewritten using a combination of [`tsconfig paths`](https://www.typescriptlang.org/tsconfig#paths) and [`tsc-alias`](https://www.npmjs.com/package/tsc-alias).
- PDKProject has been refactored to pull out PDKRelease as this should now only happen in `AwsPdkProject`.
- Package tasks have been removed from individual packages as this now should only occur in `aws-pdk`.
-`aws_arch` and `cdk_graph_plugin_diagram` have been refactored slightly to ensure exported types make `JSII` happy.
- Overall DAG has been simplified and improvements to build times have been made ~8 mins
- `docs` has been pulled up to the root level
- `private` has been replaces by `projenrc`
- `PDKPipelineXXProject` has been removed from the `pipeline` package.
- NXMonorepo ensure NodeResolution is `nodenext` for TS projects
- Changed transient deps script to use shallowest transient dep version instead of greatest given we are now bundling everything together in one.
- Interfaces no longer expose required props when they aren't actually required.
- All TSProjects under a monorepo use tsconfig module reoslution settings of `nodenext` and target `es2021`
- Added a global `pdk` command which aliases underlying package manager command 🕺 
- Added an API Explorer powered by SwaggerUI to the CloudscapeReactTsWebsite which allows you to make authenticated API requests
- Added an Infra package which allows you to create boilerplate CDK code for an API + Website. This means you can get a fully functioning website + API, with all Infra in < 70 LoC 👏 

## BREAKING CHANGES

- All imports will need to be changed to `aws-pdk/<pkg>`
- All dependencies on `@aws-prototyping-sdk/*` will need to be changed to a single dependency on `aws-pdk`
- Refactored CDKGraph & CdkGraphDiagramPlugin to poly friendly definitions which will break existing consumers in TS (Py and Java never worked).
- Removed PDKPipeline projen constructs.
- Renamed `nx-monorepo` to `monorepo`
- TypeSafeApi Python runtime moved to new `python-nextgen` OpenAPI generator, which means:
  - Properties of models can be referenced as attributes rather than dictionary syntax (eg `.my_property` rather than `["my_property"]`).
  - Models are now serialised and deserialised using the `.from_json` and `.to_json` methods.
  - `request_parameters` and `request_array_parameters` are also models.
- TypeSafeApi Python handlers import location changed from `<package>.apis.tags.default_api_operation_config` to `<package>.api.operation_config`.
- TypeSafeApi Java classes for handlers/interceptors in runtime are no longer static subclasses of the `Handlers` class. The import location has changed from eg `<package>.api.Handlers.SayHello` to `<package>.api.handlers.say_hello.SayHello`.

### Remaining items
- [ ] Update Github workflow and re-enable doc publishing once written
- [ ] Update documentation including migration guide
- [ ] Document experimental vs stable
- [x] Bump deps
- [x] Create Infra project that gens infra for API + Website (Blueprint)
- [x] Clean up MonorepoTs props to categorize yarn vs pnpm specific props
- [x] Resolve #365
- [x] Refactor cloudscape construct to use native AppLayout oppose to NorthStar AppLayout
- [x] Relax ESLINT/TSC errors in website for dev
- [x] Namespace scripts
- [x] Rename NXMonorepo
- [x] Add powertools to handlers
- [x] Resolve #527
- [x] Refactor projen constructs to remove unneeded props i.e. defaultReleaseBranch
- [x] Cherry pick @cogwirrel changes

### CODE REVIEW

File: packages/aws-pdk/src/aws_arch/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
```
Comment (cogwirrel) [inline packages/aws-pdk/src/aws_arch/src/index.ts]: I think you want to gitignore `packages/aws-pdk/src` :)

**Review by cogwirrel — COMMENTED** [review]
Looking great so far!

You'll also need to update the import in the `addDeps` for the tsapi Java/Python infra projects :)

File: packages/aws-pdk/src/aws_arch/src/index.ts (hunk 1 above)
Comment (agdimech) [reply packages/aws-pdk/src/aws_arch/src/index.ts]: Good catch - updated!

**Review by JeremyJonas — APPROVED** [review]
Awesome! Great fix :D

**Pull request merged**